File: Week4/Training/MVVMExample/app/src/main/java/com/example/admin/mvvmexample/MainActivity.java
package com.example.admin.mvvmexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.admin.mvvmexample.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity<ActivityMainBinding, UserViewModel> {

    @Override
    public UserViewModel onCreate() {
        return new UserViewModel(this);
    }

    @Override
    public int getVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }
}

File: Week7/FireBaseExample/app/src/main/java/com/example/firebaseexample/MainActivity.kt
package com.example.firebaseexample

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Log
import android.view.View
import androidx.databinding.DataBindingUtil
import com.example.firebaseexample.databinding.ActivityMainBinding
import com.example.firebaseexample.fragments.LoginFrag








class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        supportFragmentManager.beginTransaction()
                .replace(R.id.frag_container, LoginFrag())
                .commit()

    }

}
